v0/destinations/candu/transform.js

```diff
@@ -14,9 +15,23 @@ const responseBuilder = (body, { Config }) => {
   payload.context = {
     source: "RudderStack"
   };
+  if (payload.userId) {
+    payload.userId = `${payload.userId}`;
+  }
+  const revisedPayload = {};
+  // here we are trying to check if there is any empty object, array or string inside the payload.
+  for (const [key, value] of Object.entries(payload)) {
+    if (_.isObject(value) || _.isArray(value) || _.isString(value)) {
+      if (!_.isEmpty(value)) {
+        revisedPayload[key] = value;
+      }
+    } else {
+      revisedPayload[key] = value;
+    }
+  }
```

Comment:
Do we need this section? `removeUndefinedAndNullAndEmptyValues` automatically handles empty cases for `object - {}`, `array - []` and constructpayload() handles the case if empty `string - ""` is provided

Reply:
That was the issue actually. you can go through this message once -> https://rudderlabs.slack.com/archives/C02A3UAU9UN/p1644768372033719?thread_ts=1644247339.620279&cid=C02A3UAU9UN

Reply:
we now no longer require this check as it is being handled when we are converting the `userId` to string itself.